build.rs

```diff
@@ -508,14 +503,28 @@ fn build_library(
     println!("cargo:rustc-link-lib=static={}", lib_name);
 }
 
-fn compile(p: &Path, target: &Target, warnings_are_errors: bool, out_dir: &Path) -> String {
+fn compile(
+    p: &Path,
+    target: &Target,
+    compiler: &cc::Tool,
+    warnings_are_errors: bool,
+    out_dir: &Path,
+) -> String {
     let ext = p.extension().unwrap().to_str().unwrap();
     if ext == "o" {
         p.to_str().expect("Invalid path").into()
     } else {
         let out_path = obj_path(out_dir, p);
         let cmd = if target.os != WINDOWS || ext != "asm" {
-            cc(p, ext, target, warnings_are_errors, &out_path, out_dir)
+            cc(
```

Comment:
There's still a call in `pregenerate_asm_main` that calls `nasm` directly though

Reply:
> There's still a call in pregenerate_asm_main that calls nasm unconditionally though.

I'm making more changes to address that in upcoming PRs.